tests/TestCase/ORM/QueryTest.php

```diff
@@ -1328,13 +1344,17 @@ public function testFormatResultsBelongsToMany()
             $this->assertInstanceOf('Cake\ORM\Entity', $r);
         }
 
+        $this->assertTrue($first->beforeFind);
+        $this->assertNull($first->joinBeforeFind);
+
         $this->assertCount(2, $first->tags);
         $expected = [
             'id' => 1,
             'name' => 'tag1',
             '_joinData' => ['article_id' => 1, 'tag_id' => 1],
             'description' => 'A big description',
             'created' => new Time('2016-01-01 00:00'),
+            'joinBeforeFind' => true,
```

Comment:
Why is this here and not in `_joinData`? I don't understand how after being applied to `articles_tags` you expect the results to be here